### PR TITLE
Fix problem with `UnexpectedFrameEnd` raise

### PR DESCRIPTION
**Pull request by grk**

### CODE REVIEW

**Review by dentarg — APPROVED** [review]
Nice catch

**Comment by dentarg** [comment]
@grk are you able to say how you found this? did your code crash? do you know how to reproduce the situation?

**Comment by grk** [comment]
I messed up my config and tried to connect to the mqtt port of our rabbitmq instance 🤦🏻

File: lib/amqp/client/connection.rb [hunk 1]
```diff
@@ -443,7 +443,7 @@ def establish(socket, user, password, vhost, options)
 
           type, channel_id, frame_size = buf.unpack("C S> L>")
           frame_end = buf.getbyte(frame_size + 7)
-          raise UnexpectedFrameEndError, frame_end if frame_end != 206
+          raise UnexpectedFrameEnd, frame_end if frame_end != 206
```
Comment (dentarg) [inline lib/amqp/client/connection.rb:446]: Actually, this is still not correct, it should be `Error::UnexpectedFrameEnd`, noticed this when writing a test for this, I will push it to this branch soon

**Pull request merged**